test/vms/cirros/download-cirros

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: The RamenDR authors
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import json
+import logging
+import os
+import shutil
+import subprocess
+import tempfile
+
+
+def main():
+    p = argparse.ArgumentParser("download-cirros")
+    p.add_argument("--cirros-version", default="0.6.2", help="Cirros version")
+    p.add_argument("-o", "--output", help="Output filename")
+    p.add_argument("-v", "--verbose", action="store_true", help="Be more verbose")
+    args = p.parse_args()
+
+    logging.basicConfig(
+        level=logging.DEBUG if args.verbose else logging.INFO,
+        format="%(asctime)s %(levelname)-7s %(message)s",
+    )
+
+    out_img = args.output or f"cirros-{args.cirros_version}.qcow2"
+
+    with tempfile.TemporaryDirectory(prefix="download-cirros-") as tmp_dir:
+        work_img = os.path.join(tmp_dir, out_img)
+        meta_data = os.path.join(tmp_dir, "meta-data")
+        nocloud_img = os.path.join(tmp_dir, "nocloud.iso")
+
+        download_image(args.cirros_version, work_img)
+        make_nocloud_img(meta_data, nocloud_img)
+        unpack_rootfs(work_img, nocloud_img)
+        shutil.copyfile(work_img, out_img)
+
+
+def download_image(version, out):
+    logging.debug("Downloading cirros version %s to %s", version, out)
+    url = (
+        f"https://download.cirros-cloud.net/{version}/cirros-{version}-x86_64-disk.img"
+    )
+    cmd = ["curl", "--no-progress-meter", "--location", "--output", out, url]
+    logging.debug("Running %s", cmd)
+    subprocess.run(cmd, check=True)
+
+
+def make_nocloud_img(meta_data, out):
+    """
+    Create clound-init nocloud image with the required meta-data json. This
+    avoids the slow lookup for metadata server. When the image is started in
+    teh real deployment it will get a new instance-id and will be configured
+    again.
+    """
+    logging.debug("Creating cloud-init image %s", out)
+    meta = {"instance-id": "download-cirros"}
+    with open(meta_data, "w") as f:
+        f.write(json.dumps(meta) + "\n")
+    cmd = [
+        "genisoimage",
+        "-quiet",
+        "-output",
+        out,
+        "-volid",
+        "cidata",
+        "-joliet",
+        "-rock",
+        meta_data,
+    ]
```

Comment:
Can be replaced with cloud-localds used in the makefile to remove the genisoimage dependency.